packages/react-components/src/components/RTE/RichTextEditor.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as React from 'react';
+import { ContentEdit, Watermark } from 'roosterjs-editor-plugins';
+import { Editor } from 'roosterjs-editor-core';
+import { EditorOptions, IEditor } from 'roosterjs-editor-types';
+import {
+  Rooster,
+  createRibbonPlugin,
+  createUpdateContentPlugin,
+  Ribbon,
+  getButtons,
+  RibbonButton,
+  KnownRibbonButtonKey,
+  UpdateMode
+} from 'roosterjs-react';
+import { richTextEditorStyle } from '../styles/RichTextEditor.styles';
+
+/**
+ * Props for {@link RichTextEditor}.
+ *
+ * @beta
+ */
+export interface RichTextEditorProps {
+  content?: string;
+  onChange: (newValue?: string) => void;
+  children: React.ReactNode;
+  placeholderText?: string;
+}
+
+/**
+ * A component to wrap RoosterJS Rich Text Editor.
+ *
+ * @beta
+ */
+export const RichTextEditor: React.FC<RichTextEditorProps> = (props) => {
+  const { content, onChange, children, placeholderText } = props;
+  const editor = React.useRef<IEditor | null>(null);
+  const ribbonPlugin = React.useMemo(() => createRibbonPlugin(), []);
+
+  React.useEffect(() => {
+    if (content !== editor.current?.getContent()) {
+      editor.current?.setContent(content || '');
+    }
+  }, [content]);
+
+  const onRenderRibbon = function (): JSX.Element {
```

Comment:
In this case, onRenderRibbon will be created for every render and this is not a reccomended approach.
Code in this function can be moved  out of `RichTextEditor` to a separate component

Reply:
Yeah, let's investigate this in this ticket: [USER STORY 3429219](https://skype.visualstudio.com/SPOOL/_workitems/edit/3429219)

Reply:
This is an example of a "bad" code in React that will cause issues later when componet will grow and it would be more difficult to find a cause. Let's ty to avoid such situation from the beginning